corehq/apps/events/models.py

```diff
@@ -477,6 +478,24 @@ def save(self, *args, **kwargs):
         }
         helper.update(case_data)
 
+    def active_event_count(self, *args, **kwargs):
+        """
+        Returns how many events this attendee is currently being tracked in.
+        Specifically, these are events that have not started yet, or are currently in progress.
+        """
+        # Get events that haven't started yet or are in progress
+        today = date.today()
+        events = Event.objects.by_domain(self.domain).filter(
+            Q(start_date__gt=today) | Q(start_date__lte=today, end_date__gte=today)
+        )
+        active_event_count = 0
+        for event in events:
+            attendee_ids = [attendee.case_id for attendee in event.get_expected_attendees()]
+            if self.case_id in attendee_ids:
+                active_event_count += 1
```

Comment:
We only really care about a "go / no go" on whether we can delete the attendee.

Instead of looping through all the events, and for each one looping through all its attendees, in order to get a count, what about starting from the other side: get the extension cases of the attendee, and then iterate the events until you find the first active one, and then return a "no go".

That's an *O(n)* instead of an *O(n**2)*.

Reply:
In light of @Charl1996's comments, I have reworked the function for determining if an attendee is tracked. The overhead for calculating this is much lighter and shouldn't be as much of a performance concern.